modules/aws_vpn/aws-vpn.md

```diff
@@ -0,0 +1,83 @@
+---
+title: "aws-vpn"
+linkTitle: "aws-vpn"
+date: 2022-04-28
+draft: false
+weight: 1
+description: Creates an AWS VPN endpoint for folks to access your private Opta network
+---
+
+This module sets up an [AWS Client VPN Endpoint](https://aws.amazon.com/vpn/client-vpn/) for you. For those new to
+this service, AWS Client VPN is AWS' implementation of a VPN which natively integrates to their networking solutions.
+It follows the OpenVPN specification and can readily be connected to with the [OpenVPN Client](https://openvpn.net/vpn-client/)
+(which we recommend).
+
+As with other VPN solutions, this one is used to gain secure access to a private network, in this case the private VPC
+subnets Opta provisioned for your environment. This means that when connected you should have the network access needed
+to locally connect to your databases, caches, pretty much all resources in your private network that are accepting 
+traffic.
+
+## Adding VPN to your Environment
+For most cases, it's actually just adding a single line for the VPN module like so:
+```yaml
+name: testing-vpn
+org_name: runx
+providers:
+  aws:
+    region: us-east-1
+    account_id: XXXXXXXXXX
+modules:
+  - type: base
+  - type: aws-vpn
+  - type: k8s-cluster
+  - type: k8s-base
+```
+For upcoming work, it's best to place it just after the base module instead of at the end.
+Run `opta apply` and you should be good to go!
+
+## Setting Up the Client
+Short story:
+AWS VPN uses the OpenVPN client and the ovpn file format to set up the configuration for the connection profile. Opta
+makes a default profile for you and store is in an SSM parameter which you can fetch it from (refer to the 
+`ovpn_profile_parameter_arn` output). Simply fetch the value from this parameter, write it into a .ovpn file locally
+and tell your OpenVPN client to import a new profile from said file. And that's it!
+
+Long Story:
+Our VPN uses mTLS to handle the network encryption and validation of both the server and each client connecting to it.
+Opta creates a brand new Certificate Authority to provision the certs for both the servers and clients. As in larger
+companies it's an important security consideration to revoke VPN access on a per-person use case, it's highly 
+recommended to generate a new client key/cert for each user. This can be done in the following steps:
+
+1. Download the certificate and key of the CA created for your VPN. Again, we store them as SSM parameters whose 
```

Comment:
Can you clarify that steps 2-4 need to be followed for each user?
Also, what are the steps to revoke a user's certificates?